client/src/raw.rs

```diff
@@ -435,10 +478,37 @@ proxier! {
         start_workflow_execution,
         StartWorkflowExecutionRequest,
         StartWorkflowExecutionResponse,
-        |r| {
+        |r, workers| {
+            let mut slot: Option<Box<dyn Slot>> = None;
             let mut labels = AttachMetricLabels::namespace(r.get_ref().namespace.clone());
             labels.task_q(r.get_ref().task_queue.clone());
             r.extensions_mut().insert(labels);
+            let req_mut = r.get_mut();
+            if req_mut.request_eager_execution {
+                let namespace = req_mut.namespace.clone();
+                let task_queue = req_mut.task_queue.clone().unwrap().name.clone();
+                match workers.read().try_reserve_wft_slot(namespace, task_queue) {
+                    Some(s) => slot = Some(s),
+                    None => req_mut.request_eager_execution = false
+                }
+            }
+            slot
+        },
+        start_workflow_execution, // Disambiguate the macro
+        |resp, slot| {
+            if let Some(mut s) = slot {
+                if resp.is_ok() {
+                    let response_inner = resp.as_ref().unwrap().get_ref().clone();
```

Comment:
You can avoid this unwrap here, you should use another `if let`

Reply:
Much cleaner, done...